### PR TITLE
TestClusterTopology: prune TombStone tikv automatically

### LINKED ISSUE #91: TestClusterTopology: need to manually clean up the useless tikv resources after scale-in tikv
## Development Task
企业微信截图_fed778dd-ae25-480a-881e-f3a157238125
After we modify testclustertopology yaml to scale-int tikv, the resources occupied by tikv will not be cleaned up automatically. We need to manually log in to the central control computer and call prune to clean up.

### PR DESCRIPTION
**Pull request by CadmusJiang**
### What problem does this PR solve?

Issue Number: close #91  

Problem Summary:  tikv tombstone resources need to be actively pruned

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:
In ready(testclustertopology) state, reconcile check tidb cluster by calling cmd(tiup cluster display).

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
1.tikv(n4,n3,n2,n1)--> tikv (n3,n2,n1)
check n4 tikv is pruned successfully
2.tikv(n3,n2,n1)--> tikv (n2,n1)
check reconcile if try prune continuously
3.tikv(n2,n1)--->tikv(n2,n1,n4)
check n3 tikv is pruned successfully

Side effects

- Breaking backward compatibility

### CODE REVIEW

File: api/v1/testclustertopology_webhook.go [hunk 1]
```diff
@@ -108,6 +108,10 @@ func (r *TestClusterTopology) ValidateUpdate(old runtime.Object) error {
 		return fmt.Errorf("you must fill %v", result)
 	}
 
+	if IsScaleIn(tct.Status.PreTiDBCluster, r.Spec.TiDBCluster) && (len(r.Status.PendingOfflineList) != 0 || len(r.Status.OfflineList) != 0) {
+		return fmt.Errorf("you must wait tikvs %v finished", r.Status.PendingOfflineList)
```
Comment (mahjonp) [inline api/v1/testclustertopology_webhook.go:112]: What does Finish" mean?

File: controllers/testclustertopology_controller.go [hunk 2]
```diff
@@ -124,9 +124,26 @@ func (r *TestClusterTopologyReconciler) Reconcile(req ctrl.Request) (ctrl.Result
...
 		// cluster config no change
 		if !tiup.IsClusterConfigModified(ct.Status.PreTiDBCluster, ct.Spec.TiDBCluster) {
+			var rr naglfarv1.TestResourceRequest
+			if err := r.Get(ctx, types.NamespacedName{
+				Namespace: req.Namespace,
+				Name:      ct.Spec.ResourceRequest,
+			}, &rr); err != nil {
+				return ctrl.Result{}, err
+			}
+			isRequeue, result, err := r.pruneTiDBCluster(ctx, &ct, &rr)
```
Comment (mahjonp) [inline controllers/testclustertopology_controller.go:136]: Source viewers(like me) may don't know why needs do `pruneTiDBCluster` here. Perhaps you should create a new function, or add an if-condition (predicate according to `status` that need to wait for prune finished) around this code or give it a not too bad name to it.

**Review by mahjonp — COMMENTED** [review]
Please take a look @CadmusJiang and I think we should begin to add unit tests on each PR.

**Review by Hexilee — APPROVED** [review]
LGTM

**Review by mahjonp — APPROVED** [review]
LGTM

**Pull request merged**